### PR TITLE
Refactor: Re enable and fix Ruff's TRY Checks

### PR DESCRIPTION
**Pull request by ELC**
## What type of PR is this? (check all applicable)

- [X] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Checklist

- [X] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

### CODE REVIEW

**Review by FBruzzesi — COMMENTED** [review]
Thanks for the PR!

In the code blocks where TRY003 and TRY004 are satisfied we currently use `msg = ... `. 
I would rather aim to keep it the same variable name for the entire codebase. 

Personally `error_message` is a bit lengthy, while `msg` may be too generic.

Maybe `err_msg` is the compromise we can settle with?

Edit: I just noticed #447 that renames all the other instances

**Comment by MarcoGorelli** [comment]
yeah let's get the other one in first

File: narwhals/_pandas_like/group_by.py [hunk 1]
```diff
@@ -146,9 +146,10 @@ def agg_pandas(  # noqa: PLR0913
         try:
             result_simple = grouped.agg(aggs)
         except AttributeError as exc:
-            raise RuntimeError(
+            error_message = (
                 "Failed to aggregated - does your aggregation function return a scalar?"
-            ) from exc
+            )
+            raise RuntimeError(error_message) from exc
```
Comment (MarcoGorelli) [inline narwhals/_pandas_like/group_by.py:152]: this is fine, happy to add this (with `msg` instead of `error_message`)

**Comment by ELC** [comment]
Changed all instances to `msg`

I do agree with FBruzzesi and would avoid variable names that are too short, but this is not critical

**Review by FBruzzesi — APPROVED** [review]
Thanks for adjusting 👌

KittyKittenGIF

**Pull request merged**